1주차/3.웨비의 사진관/style.css

```
@@ -1,159 +1,265 @@
+html {
+  scroll-behavior: smooth;
+}
+
 body {
-    font-weight: bold;
+  font-weight: bold;
+  background-color: #ecffdc;
 }
 
 a {
-    text-decoration-line: none;
+  text-decoration-line: none;
 }
 
-a:hover { color:mediumaquamarine; text-decoration-line: underline; }
-
-.header1 {
-    display: flex;
-    align-items: center;
-    background-color: mediumaquamarine;
-    height: 40px;
+a:hover {
+  color: mediumaquamarine;
+  text-decoration-line: underline;
+}
 
+ul {
+  display: flex;
+  justify-content: center;
+  padding-left: 0;
+  list-style: none;
 }
 
-.header {
-    display: flex;
-    align-items: center;
-    background-color: mediumaquamarine;
-    height: 40px;
-    position: sticky;
-    top: 0; /* 화면 상단에 고정 */
+li {
+  padding: 0 0.5rem;
+}
 
+.main-title {
+  display: flex;
+  align-items: center;
+  padding-left: 1rem;
+  background-color: mediumaquamarine;
+  height: 3rem;
 }
 
+.country {
+  display: flex;
+  align-items: center;
+  padding-left: 1rem;
+  background-color: mediumaquamarine;
+  height: 3rem;
+  position: sticky;
+  top: 0; /* 화면 상단에 고정 */
+}
 
 .img-preview {
-    display: flex;
-    overflow-x: auto; /*가로 스크롤바*/
-    height: 450px;
-    align-items: center;
+  display: flex;
+  overflow-x: auto; /*가로 스크롤바*/
+  margin: 0 1rem;
+  height: 28rem;
+  align-items: center;
+  position: relative;
+}
+
+::-webkit-scrollbar {
+  width: 1rem;
+}
+
+.img-preview::-webkit-scrollbar-thumb {
+  background-color: mediumaquamarine;
+  border-radius: 3rem;
 }
 
 .img-preview img {
-    width: 300px;
-    height: 400px;
-    margin-left: 5px;
-    margin-right: 5px;
+  width: 20rem;
+  height: 25rem;
+  margin-left: 0.5rem;
+  margin-right: 0.5rem;
 }
 
 .nav {
-    display: flex;
-    justify-content: center;
-    align-items: center;
-    flex-direction: column;
-    height: 100px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  flex-direction: column;
+  height: 7rem;
 }
 
 .img-france {
-    display: flex;
-    flex-wrap: wrap;
-    height: 450px;
-    align-items: center;
-    justify-content: space-around;
+  display: flex;
+  flex-wrap: wrap;
+  padding: 1rem 3.5rem;
+  gap: 1rem;
+  height: auto;
+  align-items: center;
+  flex-grow: 1;
+  justify-content: center;
+}
+.each {
+  position: relative; /*필요 */
+  width: 14rem;
+  height: 14rem;
+  flex-grow: 1;
+  text-align: center;
+  object-fit: cover;
 }
 
 .img-france img {
-    width: 17%;
-    height: 47%;
-    margin-left: 5px;
-    margin-right: 5px;
+  width: 100%;
+  height: 100%;
+  object-fit: cover;
 }
 
-.img-portugal {
-    display: flex;
-    height: 450px;
-    margin: 0 auto;
-    justify-content: center;
+.hover-text {
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  width: 100%;
+  transform: translate(-50%, -50%); /* 텍스트를 가운데로 이동 */
```

나는 텍스트를 가운데로 이동시킬때는 이런 식으로 써주곤 해!
```suggestion
  text-align: center; /* 텍스트를 가운데로 이동 */
```

혹시 text-align이나 justify-content가 아닌, transform을 활용한 이유를 알 수 있을까?
보통의 경우에는 텍스트 가운데 정렬에 transform 보단 `text-align`이나 `justify-content`를 많이 사용한다고 느꼈거든!

---

` transform: translate(-50%, -50%);` 가 없으면 변환하는 요소의 중심이 왼쪽 상단이야!! 저걸 추가해서 난 요소의 중심을 딱 중앙으로 맞춘거양!! 그러면 `top:50%; left:50%` 설정함으로써 text가 딱 중앙으로 와
만약 height도 100%로 하면 text 세로 중앙정렬하기 위해 `display:flex;, justify-content:center` 등등의 설정을 해야겠지만 난 height은 따로 설정 안했고,  transform을 해야겠다는 생각이 먼저 들어서 저렇게 했어!